system/lib/wasm_worker/library_wasm_worker.c

```diff
@@ -21,8 +22,18 @@ void __wasm_init_tls(void *memory);
 __attribute__((constructor(48)))
 static void emscripten_wasm_worker_main_thread_initialize() {
 	uintptr_t* sbrk_ptr = emscripten_get_sbrk_ptr();
-	__wasm_init_tls((void*)*sbrk_ptr);
-	*sbrk_ptr += __builtin_wasm_tls_size();
+	uintptr_t sbrk_addr = *sbrk_ptr;
+
+	if(__builtin_wasm_tls_align() != 0)
+	{
+		sbrk_addr = (sbrk_addr + __builtin_wasm_tls_align() - 1) & (-__builtin_wasm_tls_align());
```

Comment:
How about using a macro like in pthread_create.c? (`#define ROUND_UP(x, ALIGNMENT) (((x)+ALIGNMENT-1)&-ALIGNMENT`)